Module-path tests fail silently

The module-path tests (Java 9+) fail but they are silently ignored. We didn't realize it before because of the test jobs run with `-Dmaven.test.failure.ignore=true`.

The output from PR builder [job 650](http://jenkins.hazelcast.com/view/PR%20Builders/job/Hazelcast-pr-builder/650/consoleFull):
```
13:34:52 [ERROR] There are test failures.
13:34:52 
13:34:52 Please refer to /home/jenkins/jenkins_slave/workspace/Hazelcast-pr-builder/modulepath-tests/target/surefire-reports for the individual test results.
13:34:52 Please refer to dump files (if any exist) [date]-jvmRun[N].dump, [date].dumpstream and [date]-jvmRun[N].dumpstream.
13:34:52 The forked VM terminated without properly saying goodbye. VM crash or System.exit called?
13:34:52 Command was /bin/sh -c cd /home/jenkins/jenkins_slave/workspace/Hazelcast-pr-builder/modulepath-tests && /usr/share/jdks/jdk-11/bin/java -Xms512m -Xmx2G -XX:MaxPermSize=1024M --add-opens jdk.management/com.sun.management.internal=com.hazelcast.core --add-opens java.base/sun.nio.ch=com.hazelcast.core --add-opens java.base/java.lang=com.hazelcast.core --add-opens java.management/sun.management=com.hazelcast.core --add-exports java.xml/jdk.xml.internal=com.hazelcast.core --illegal-access=warn -Dhazelcast.phone.home.enabled=false -Dlog4j.skipJansi=true @/home/jenkins/jenkins_slave/workspace/Hazelcast-pr-builder/modulepath-tests/target/surefire/surefireargs14615101039161651973 /home/jenkins/jenkins_slave/workspace/Hazelcast-pr-builder/modulepath-tests/target/surefire 2019-03-01T12-34-51_335-jvmRun1 surefire6974081814789914269tmp surefire_010893843922274902116tmp
13:34:52 Error occurred in starting fork, check output in log
13:34:52 Process Exit Code: 1
13:34:52 org.apache.maven.surefire.booter.SurefireBooterForkException: The forked VM terminated without properly saying goodbye. VM crash or System.exit called?
13:34:52 Command was /bin/sh -c cd /home/jenkins/jenkins_slave/workspace/Hazelcast-pr-builder/modulepath-tests && /usr/share/jdks/jdk-11/bin/java -Xms512m -Xmx2G -XX:MaxPermSize=1024M --add-opens jdk.management/com.sun.management.internal=com.hazelcast.core --add-opens java.base/sun.nio.ch=com.hazelcast.core --add-opens java.base/java.lang=com.hazelcast.core --add-opens java.management/sun.management=com.hazelcast.core --add-exports java.xml/jdk.xml.internal=com.hazelcast.core --illegal-access=warn -Dhazelcast.phone.home.enabled=false -Dlog4j.skipJansi=true @/home/jenkins/jenkins_slave/workspace/Hazelcast-pr-builder/modulepath-tests/target/surefire/surefireargs14615101039161651973 /home/jenkins/jenkins_slave/workspace/Hazelcast-pr-builder/modulepath-tests/target/surefire 2019-03-01T12-34-51_335-jvmRun1 surefire6974081814789914269tmp surefire_010893843922274902116tmp
13:34:52 Error occurred in starting fork, check output in log
13:34:52 Process Exit Code: 1
13:34:52 	at org.apache.maven.plugin.surefire.booterclient.ForkStarter.fork(ForkStarter.java:671)
13:34:52 	at org.apache.maven.plugin.surefire.booterclient.ForkStarter.fork(ForkStarter.java:533)
13:34:52 	at org.apache.maven.plugin.surefire.booterclient.ForkStarter.run(ForkStarter.java:278)
13:34:52 	at org.apache.maven.plugin.surefire.booterclient.ForkStarter.run(ForkStarter.java:244)
13:34:52 	at org.apache.maven.plugin.surefire.AbstractSurefireMojo.executeProvider(AbstractSurefireMojo.java:1194)
13:34:52 	at org.apache.maven.plugin.surefire.AbstractSurefireMojo.executeAfterPreconditionsChecked(AbstractSurefireMojo.java:1022)
13:34:52 	at org.apache.maven.plugin.surefire.AbstractSurefireMojo.execute(AbstractSurefireMojo.java:868)
13:34:52 	at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:137)
13:34:52 	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:208)
13:34:52 	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:154)
13:34:52 	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:146)
13:34:52 	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:117)
13:34:52 	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:81)
13:34:52 	at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:56)
13:34:52 	at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:128)
13:34:52 	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:305)
13:34:52 	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:192)
13:34:52 	at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:105)
13:34:52 	at org.apache.maven.cli.MavenCli.execute(MavenCli.java:954)
13:34:52 	at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:288)
13:34:52 	at org.apache.maven.cli.MavenCli.main(MavenCli.java:192)
13:34:52 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
13:34:52 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
13:34:52 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
13:34:52 	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
13:34:52 	at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:289)
13:34:52 	at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:229)
13:34:52 	at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:415)
13:34:52 	at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:356)
```

The underlying problem seems to be this one:
```
# Created at 2019-03-01T15:09:58.960
Corrupted STDOUT by directly writing to native stream in forked JVM 1. Stream 'Caused by: java.lang.module.InvalidModuleDescriptorException: Provider class com.fasterxml.jackson.core.JsonFactory not in module'.
java.lang.IllegalArgumentException: Stream stdin corrupted. Expected comma after third character in command 'Caused by: java.lang.module.InvalidModuleDescriptorException: Provider class com.fasterxml.jackson.core.JsonFactory not in module'.
	at org.apache.maven.plugin.surefire.booterclient.output.ForkClient$OperationalData.<init>(ForkClient.java:511)
	at org.apache.maven.plugin.surefire.booterclient.output.ForkClient.processLine(ForkClient.java:209)
	at org.apache.maven.plugin.surefire.booterclient.output.ForkClient.consumeLine(ForkClient.java:176)
	at org.apache.maven.plugin.surefire.booterclient.output.ThreadedStreamConsumer$Pumper.run(ThreadedStreamConsumer.java:88)
	at java.base/java.lang.Thread.run(Thread.java:834)
```

--- a/hazelcast/pom.xml
+++ b/hazelcast/pom.xml
@@ -213,7 +213,7 @@
             <plugin>
                 <groupId>org.apache.maven.plugins</groupId>
                 <artifactId>maven-shade-plugin</artifactId>
-                <version>2.3</version>
+                <version>${maven.shade.plugin.version}</version>
                 <executions>
                     <execution>
                         <phase>package</phase>
@@ -242,6 +242,9 @@
                                     <shadedPattern>com.hazelcast.org.snakeyaml</shadedPattern>
                                 </relocation>
                             </relocations>
+                            <transformers>
+                                <transformer implementation="org.apache.maven.plugins.shade.resource.ServicesResourceTransformer"/>
+                            </transformers>
                         </configuration>
                     </execution>
                 </executions>

--- a/pom.xml
+++ b/pom.xml
@@ -62,7 +62,7 @@
         <maven.assembly.plugin.version>2.4</maven.assembly.plugin.version>
         <maven.rar.plugin.version>2.2</maven.rar.plugin.version>
         <maven.bundle.plugin.version>2.4.0</maven.bundle.plugin.version>
-        <maven.shade.plugin.version>2.2</maven.shade.plugin.version>
+        <maven.shade.plugin.version>3.0.0</maven.shade.plugin.version>
         <maven.dependency.plugin.version>2.10</maven.dependency.plugin.version>
         <maven.animal.sniffer.plugin.version>1.15</maven.animal.sniffer.plugin.version>
         <maven.git.commit.id.plugin.version>2.1.10</maven.git.commit.id.plugin.version>

